BookingsApi/BookingsApi/Controllers/V1/HearingsController.cs

```diff
@@ -213,6 +213,58 @@ public async Task<IActionResult> UpdateHearingsInGroup(Guid groupId, [FromBody]
             return NoContent();
         }
 
+        /// <summary>
+        /// Cancel hearings in a multi day group
+        /// </summary>
+        /// <returns>No content</returns>
+        [HttpPatch("{groupId}/hearings/cancel")]
+        [OpenApiOperation("CancelHearingsInGroup")]
+        [ProducesResponseType((int)HttpStatusCode.NoContent)]
+        [ProducesResponseType(typeof(ValidationProblemDetails), (int)HttpStatusCode.BadRequest)]
+        [ProducesResponseType((int)HttpStatusCode.NotFound)]
+        [MapToApiVersion("1.0")]
+        public async Task<IActionResult> CancelHearingsInGroup(Guid groupId, [FromBody] CancelHearingsInGroupRequest request)
+        {
+            var inputValidationResult = await new CancelHearingsInGroupRequestInputValidation().ValidateAsync(request);
+            if (!inputValidationResult.IsValid)
+            {
+                ModelState.AddFluentValidationErrors(inputValidationResult.Errors);
+                return ValidationProblem(ModelState);
+            }
+            
+            var getHearingsByGroupIdQuery = new GetHearingsByGroupIdQuery(groupId);
+            var hearingsInGroup = await _queryHandler.Handle<GetHearingsByGroupIdQuery, List<VideoHearing>>(getHearingsByGroupIdQuery);
+            
+            if (!hearingsInGroup.Any())
+            {
+                return NotFound();
+            }
+            
+            var dataValidationResult = await new CancelHearingsInGroupRequestRefDataValidation(hearingsInGroup).ValidateAsync(request);
+            if (!dataValidationResult.IsValid)
+            {
+                ModelState.AddFluentValidationErrors(dataValidationResult.Errors);
+                return ValidationProblem(ModelState);
+            }
+            
+            var requestHearings = hearingsInGroup.Where(h => request.HearingIds.Contains(h.Id)).ToList();
```

Comment:
Is it worth to put this in the service so it can be re-used?

Reply:
I don't think so personally. It's very specific to this particular API endpoint